astropy/convolution/tests/test_convolve.py

```diff
@@ -925,3 +927,32 @@ def test_non_square_kernel_asymmetric(boundary):
     image[6, 6] = 1
     result = convolve(image, kernel, normalize_kernel=False, boundary=boundary)
     assert_allclose(result[5:8, 4:9], kernel)
+
+
+@pytest.mark.parametrize(('boundary', 'normalize_kernel'),
+                         itertools.product(BOUNDARY_OPTIONS,
+                                           NORMALIZE_OPTIONS))
+def test_uninterpolated_nan_regions(boundary, normalize_kernel):
+    #8086
+    # Test NaN interpolation of contiguous NaN regions with kernels of size
+    # identical and greater than that of the region of NaN values.
+
+    # Test case: kernel.shape == NaN_region.shape
+    kernel = Gaussian2DKernel(1, 5, 5)
+    nan_centroid = np.full(kernel.shape, np.nan)
+    image = np.pad(nan_centroid, pad_width=kernel.shape[0]*2, mode='constant',
+                   constant_values=1)
+    try:
```

Comment:
Use something like this:
https://github.com/astropy/astropy/blob/master/astropy/convolution/tests/test_convolve_fft.py#L71
instead, i.e., check that the appropriate warning is raised